ledger/ledger-api-bench-tool/src/main/scala/com/daml/ledger/api/benchtool/submission/FooCommandGenerator.scala

```diff
@@ -0,0 +1,206 @@
+// Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ledger.api.benchtool.submission
+
+import com.daml.ledger.api.benchtool.config.WorkflowConfig.FooSubmissionConfig
+import com.daml.ledger.api.v1.commands.{Command, ExerciseByKeyCommand}
+import com.daml.ledger.api.v1.value.{Identifier, Record, RecordField, Value}
+import com.daml.ledger.client.binding.Primitive
+import com.daml.ledger.test.model.Foo._
+
+import java.nio.charset.StandardCharsets
+import java.util.concurrent.atomic.AtomicLong
+import scala.util.{Failure, Try}
+import scala.util.control.NonFatal
+
+case class FooTemplateDescriptor(
+    templateId: Identifier,
+    consumingChoiceName: String,
+    nonconsumingChoiceName: String,
+)
+
+/** NOTE: Keep me in sync with `Foo.daml`
+  */
+object FooTemplateDescriptor {
+
+  val Foo1: FooTemplateDescriptor = FooTemplateDescriptor(
+    templateId = com.daml.ledger.test.model.Foo.Foo1.id.asInstanceOf[Identifier],
+    consumingChoiceName = "Foo1_ConsumingChoice",
+    nonconsumingChoiceName = "Foo1_NonconsumingChoice",
+  )
+  val Foo2: FooTemplateDescriptor = FooTemplateDescriptor(
+    templateId = com.daml.ledger.test.model.Foo.Foo2.id.asInstanceOf[Identifier],
+    consumingChoiceName = "Foo2_ConsumingChoice",
+    nonconsumingChoiceName = "Foo2_NonconsumingChoice",
+  )
+  val Foo3: FooTemplateDescriptor = FooTemplateDescriptor(
+    templateId = com.daml.ledger.test.model.Foo.Foo3.id.asInstanceOf[Identifier],
+    consumingChoiceName = "Foo3_ConsumingChoice",
+    nonconsumingChoiceName = "Foo3_NonconsumingChoice",
+  )
+}
+
+final class FooCommandGenerator(
```

Comment:
Can we have a more descriptive name? At first glance, `FooCommandGenerator` makes me think I'm looking at unit test code.

Reply:
At the moment this is named badly but consistently with the `Foo.daml` definitions of the `Foo` templates and choices.  Happy to see that renamed but it would be best to leave that to @pbatko-da as he is currently making changes in that area.

Reply:
I'd retain the old name and consider changing it in a PR that introduces another flavor of command generation.

Reply:
In my latest commit I've split the files above with history preserved.  Once that is done we'll be able to rename again without he worry about preserving git history.